src/main/java/net/sf/jabref/logic/importer/fileformat/BibTeXMLImporter.java

```diff
@@ -79,33 +98,184 @@ public ParserResult importDatabase(BufferedReader reader) throws IOException {
 
         List<BibEntry> bibItems = new ArrayList<>();
 
-        // Obtain a factory object for creating SAX parsers
-        SAXParserFactory parserFactory = SAXParserFactory.newInstance();
-        // Configure the factory object to specify attributes of the parsers it
-        // creates
-        // parserFactory.setValidating(true);
-        parserFactory.setNamespaceAware(true);
-        // Now create a SAXParser object
-
         try {
-            SAXParser parser = parserFactory.newSAXParser(); //May throw exceptions
-            BibTeXMLHandler handler = new BibTeXMLHandler();
-            // Start the parser. It reads the file and calls methods of the handler.
-            parser.parse(new InputSource(reader), handler);
-            // When you're done, report the results stored by your handler object
-            bibItems.addAll(handler.getItems());
-
-        } catch (javax.xml.parsers.ParserConfigurationException e) {
+            JAXBContext context = JAXBContext.newInstance("net.sf.jabref.importer.fileformat.bibtexml");
+            XMLInputFactory xmlInputFactory = XMLInputFactory.newFactory();
+            XMLStreamReader xmlReader = xmlInputFactory.createXMLStreamReader(reader);
+
+            //go to the root element
+            while (!xmlReader.isStartElement()) {
+                xmlReader.next();
+            }
+
+            Unmarshaller unmarshaller = context.createUnmarshaller();
+            File file = (File) unmarshaller.unmarshal(xmlReader);
+
+            List<Entry> entries = file.getEntry();
+            Map<String, String> fields = new HashMap<>();
+
+            for (Entry entry : entries) {
+                BibEntry bibEntry = new BibEntry(DEFAULT_BIBTEXENTRY_ID);
+                if (entry.getArticle() != null) {
+                    bibEntry.setType("article");
+                    parse(entry.getArticle(), fields);
+                }
+                if (entry.getBook() != null) {
```

Comment:
Why not `} else if (`? Isn't it the case, that is is either article, or book, or booklet, etc?